### PR TITLE
Sqlschema Fix for views and Bake

### PR DESCRIPTION
**Pull request by ghost**
<b>In response to ticket 6501</b> ~ https://github.com/cakephp/cakephp/issues/6501

Fix for Sqlserver schema to use with MSSQL bake and views

### CODE REVIEW

File: src/Database/Schema/SqlserverSchema.php [hunk 1]
```diff
@@ -12,26 +13,47 @@
...
 namespace Cake\Database\Schema;
 
+use Cake\Database\Schema\Table;
+
 /**
  * Schema management/reflection features for SQLServer.
  */
-class SqlserverSchema extends BaseSchema
-{
+class SqlserverSchema extends BaseSchema {
```
Comment (dereuromark) [inline src/Database/Schema/SqlserverSchema.php]: You need to minimize the noise in this PR, only real changes should be commited, also your IDE seems to be misconfigured.

Comment (Xymanek) [inline src/Database/Schema/SqlserverSchema.php]: If you use PhpStorm (which I presume you do as it has a tendency to adapt all code to your editing style), I suggest you use "PSR-2" code style preset *in project scope* as it closely matches Cake's coding standard and then Code->Reformat in order to bring everything up to new coding style
screenshot_1

If you don't use PhpStorm, forgive me for off-topic tip

File: src/Database/Schema/SqlserverSchema.php [hunk 2]
```diff
@@ -12,26 +13,47 @@
  * @since         3.0.0
  * @license       http://www.opensource.org/licenses/mit-license.php MIT License
  */
+/**
```
Comment (ADmad) [inline src/Database/Schema/SqlserverSchema.php]: Lines 16 to 31 are duplicate or docblock which already exists above.

Comment (dereuromark) [inline src/Database/Schema/SqlserverSchema.php]: Yes I would keep the original header of the file.

File: src/Database/Schema/SqlserverSchema.php [hunk 3]
```diff
@@ -86,23 +106,23 @@ protected function _convertColumn($col, $length = null, $precision = null, $scal
         }
 
         if ($col === 'int' || $col === 'integer') {
-            return ['type' => 'integer', 'length' => $precision ?: 10];
+            return ['type' => 'integer', 'length' => $precision ? : 10];
```
Comment (dereuromark) [inline src/Database/Schema/SqlserverSchema.php]: Still wrong changes in here.

File: src/Database/Schema/SqlserverSchema.php [hunk 4]
```diff
@@ -144,19 +164,16 @@ protected function _convertColumn($col, $length = null, $precision = null, $scal
...
-            $row['precision'],
-            $row['scale']
+                $row['type'], $row['char_length'], $row['precision'], $row['scale']
         );
-        if (!empty($row['default'])) {
+        if ( ! empty($row['default'])) {
             $row['default'] = trim($row['default'], '()');
         }
-        if (!empty($row['autoincrement'])) {
+        if ( ! empty($row['autoincrement'])) {
```
Comment (dereuromark) [inline src/Database/Schema/SqlserverSchema.php]: Also wrong changes. You must really only change what your fix intends to change.

Comment (tgr0ss) [inline src/Database/Schema/SqlserverSchema.php]: Totally understand, it's my fault here.  My formatting was messing that up ;/   I will fix accordingly.

**Comment by markstory** [comment]
Was the union not necessary?

**Comment by tgr0ss** [comment]
Updated to the latest version that i'm using, hopefully this helps.

**Comment by tgr0ss** [comment]
@markstory -  Please let me know if there's anything else that I can do to help on this issue.

**Comment by markstory** [comment]
@tgr0ss I'm going to get this merged in tonight.

**Comment by markstory** [comment]
Rebased and merged in 3167af2cb46000115faa3200cb6de4a1e81e8576

**Pull request closed without merging**